OpenSim/Common/Component.cpp

```diff
@@ -192,6 +192,48 @@ void Component::finalizeFromProperties()
 
     markPropertiesAsSubcomponents();
     componentsFinalizeFromProperties();
+
+    auto subs = getImmediateSubcomponents();
+    std::set<std::string> names{};
+    std::string name{};
+
+    // increment the count of duplicates to use as a unique suffix
+    int count{ 0 };
+    // temp variable to hold the unique name used to rename a duplicate
+    std::string uniqueName{};
+
+    for (auto& sub : subs) {
+        name = sub->getName();
```

Comment:
This copy is unnecessary; `getName()` returns a `const string&`.

Reply:
Good find.